bee-api/bee-rest-api/src/endpoints/mod.rs

```diff
@@ -72,47 +114,64 @@ where
     }
 
     async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
-        let rest_api_config = config.0;
-        let protocol_config = config.1;
-        let network_id = config.2;
-        let bech32_hrp = config.3;
-
-        let consensus_worker = node.worker::<ConsensusWorker>().unwrap().tx.clone();
-        let tangle = node.resource::<Tangle<N::Backend>>();
-        let storage = node.storage();
-        let message_submitter = node.worker::<MessageSubmitterWorker>().unwrap().tx.clone();
-        let message_requester = node.worker::<MessageRequesterWorker>().unwrap().clone();
-        let requested_messages = node.resource::<RequestedMessages>();
-        let peer_manager = node.resource::<PeerManager>();
-        let network_controller = node.resource::<NetworkCommandSender>();
-        let node_info = node.info();
-        let bus = node.bus();
+        let args = Arc::new(ApiArgsFullNode {
```

Comment:
Have you considered:
```rust
#[derive(Clone)]
struct ApiArgsFullNode(Arc<ApiArgsFullNodeInner>);
```
to make the API a bit nicer to use?

Reply:
yea, I'ld probably do that :+1:

Reply:
Implemented it this way:

```
pub struct ApiArgsFullNode<B: StorageBackend>(Arc<ApiArgsFullNodeInner<B>>);

impl<B: StorageBackend> Deref for ApiArgsFullNode<B> {
    type Target = Arc<ApiArgsFullNodeInner<B>>;

    fn deref(&self) -> &Self::Target {
        &self.0
    }
}

impl<B: StorageBackend> Clone for ApiArgsFullNode<B> {
    fn clone(&self) -> Self {
        Self(self.0.clone())
    }
}
```